### ISSUE TITLE
From Getting Started guide, create-app looks like it works but running yarn dev gives fatal error

### ISSUE THREAD

**Issue by mmacdonald86** [backstage/backstage#14476]
### 📜 Description

Following steps in Getting Started guide, the npx @backstage/create-app command seems to run successfully to completion.  Here is the last few lines of output.
```

  copying       Root.tsx ✔
  copying       index.ts ✔
  copying       EntityPage.tsx ✔
  copying       SearchPage.tsx ✔

 Moving to final location:
  moving        backstage-dev ✔

 Installing dependencies:
  executing     yarn install ✔
  executing     yarn tsc ✔

🥇  Successfully created backstage-dev


 All set! Now you might want to:
  Run the app: cd backstage-dev && yarn dev
  Set up the software catalog: https://backstage.io/docs/features/software-catalog/configuration
  Add authentication: https://backstage.io/docs/auth/
```
  When I follow the instruction to start the server, I get fatal: not a git respository errors; it does say webpack compiled but the web server does not appear to be running:  When I try to load from http://localhost:3000, it connects but sends no response.
  
  ```
# cd backstage-dev && yarn dev
yarn run v1.22.19
$ concurrently "yarn start" "yarn start-backend"
$ yarn workspace app start
$ yarn workspace backend start
$ backstage-cli package start
$ backstage-cli package start
[1] Build succeeded
[0] Loaded config from app-config.yaml
[0] fatal: not a git repository (or any of the parent directories): .git
[0] WARNING: Failed to read git commit, ExitCodeError: Command 'git rev-parse HEAD' exited with code 128
[0] fatal: not a git repository (or any of the parent directories): .git
[0] WARNING: Failed to describe git version, ExitCodeError: Command 'git describe --always' exited with code 128
[0] <i> [webpack-dev-server] Project is running at:
[0] <i> [webpack-dev-server] Loopback: http://localhost:3000/, http://127.0.0.1:3000/
[0] <i> [webpack-dev-server] Content not from webpack is served from '/root/backstage-dev/packages/app/public' directory
[0] <i> [webpack-dev-server] 404s will fallback to '/index.html'
[1] 2022-11-07T03:27:18.521Z backstage info Loaded config from app-config.yaml
[1] 2022-11-07T03:27:18.540Z backstage info 1 secret found in the config which will be redacted
[1] 2022-11-07T03:27:18.559Z backstage info Created UrlReader predicateMux{readers=azure{host=dev.azure.com,authed=false},bitbucketCloud{host=bitbucket.org,authed=false},github{host=github.com,authed=false},gitlab{host=gitlab.com,authed=false},awsS3{host=amazonaws.com,authed=false},fetch{}
[1] 2022-11-07T03:27:18.582Z catalog info Performing database migration type=plugin
[1] 2022-11-07T03:27:18.731Z auth info Configuring "database" as KeyStore provider type=plugin
[1] 2022-11-07T03:27:18.743Z techdocs info Creating Local publisher for TechDocs type=plugin
[1] 2022-11-07T03:27:18.748Z proxy info [HPM] Proxy created: /test  -> https://example.com type=plugin
[1] 2022-11-07T03:27:18.749Z proxy info [HPM] Proxy rewrite rule created: "^/api/proxy/test/?" ~> "/" type=plugin
[1] 2022-11-07T03:27:18.750Z search info Added DefaultCatalogCollatorFactory collator factory for type software-catalog type=plugin
[1] 2022-11-07T03:27:18.750Z search info Added DefaultTechDocsCollatorFactory collator factory for type techdocs type=plugin
[1] 2022-11-07T03:27:18.751Z search info Starting all scheduled search tasks. type=plugin
[1] 2022-11-07T03:27:18.759Z app warn Can't serve static app content from /root/backstage-dev/packages/app/dist/static, directory doesn't exist type=plugin
[1] 2022-11-07T03:27:18.761Z backstage info Initializing http server
[1] 2022-11-07T03:27:18.763Z backstage info Listening on :7007
[1] 2022-11-07T03:27:18.776Z search info Task worker starting: search_index_software_catalog, {"version":2,"cadence":"PT10M","initialDelayDuration":"PT3S","timeoutAfterDuration":"PT15M"} type=taskManager task=search_index_software_catalog
[1] 2022-11-07T03:27:18.777Z search info Task worker starting: search_index_techdocs, {"version":2,"cadence":"PT10M","initialDelayDuration":"PT3S","timeoutAfterDuration":"PT15M"} type=taskManager task=search_index_techdocs
[1] 2022-11-07T03:27:21.780Z search info Collating documents for software-catalog via DefaultCatalogCollatorFactory type=plugin documentType=software-catalog
[1] 2022-11-07T03:27:21.791Z search info Collating documents for techdocs via DefaultTechDocsCollatorFactory type=plugin
 documentType=techdocs
[1] 2022-11-07T03:27:21.809Z backstage info ::ffff:127.0.0.1 - - [07/Nov/2022:03:27:21 +0000] "GET /api/catalog/entities?filter=metadata.annotations.backstage.io%2Ftechdocs-ref&fields=kind,namespace,metadata.annotations,metadata.name,metadata.title,metadata.namespace,spec.type,spec.lifecycle,relations&offset=0&limit=500 HTTP/1.1" 200 2 "-" "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)" type=incomingRequest
[1] 2022-11-07T03:27:21.815Z search info Collating documents for techdocs succeeded type=plugin documentType=techdocs
[1] 2022-11-07T03:27:21.817Z backstage info ::ffff:127.0.0.1 - - [07/Nov/2022:03:27:21 +0000] "GET /api/catalog/entities?offset=0&limit=500 HTTP/1.1" 200 - "-" "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)" type=incomingRequest
[1] 2022-11-07T03:27:21.828Z search info Collating documents for software-catalog succeeded type=plugin documentType=software-catalog
[0] webpack compiled successfully
```

  

### 👍 Expected behavior

After the yarn dev instruction says "webpack compiled successfully" I expect to be able to connect to the server at http://localhost:3000 and see the initial page

### 👎 Actual Behavior with Screenshots

Instead I get an error from Chrome:
```
This page isn’t working
localhost didn’t send any data.
ERR_EMPTY_RESPONSE
```
I suspect this might be related to the "Fatal error:" I was getting from the "yarn dev" command.

### 👟 Reproduction steps

Reproduction steps are to follow the Getting Started guide.  Which leads me to think it is a problem with my environment, but not sure how.

### 🖥️ Your Environment

```
yarn run v1.22.19
$ /root/backstage-dev/node_modules/.bin/backstage-cli info
OS:   Linux 5.10.109-104.500.amzn2.x86_64 - linux/x64
node: v16.18.1
yarn: 1.22.19
cli:  0.20.0 (installed)
backstage:  1.7.1

Dependencies:
  @backstage/app-defaults                          1.0.7
  @backstage/backend-common                        0.15.2
  @backstage/backend-plugin-api                    0.1.3
  @backstage/backend-tasks                         0.3.6
  @backstage/catalog-client                        1.1.1
  @backstage/catalog-model                         1.1.2
  @backstage/cli-common                            0.1.10
  @backstage/cli                                   0.20.0
  @backstage/config-loader                         1.1.5
  @backstage/config                                1.0.3
  @backstage/core-app-api                          1.1.1
  @backstage/core-components                       0.11.2
  @backstage/core-plugin-api                       1.0.7
  @backstage/errors                                1.1.2
  @backstage/integration-react                     1.1.5
  @backstage/integration                           1.3.2
  @backstage/plugin-api-docs                       0.8.10
  @backstage/plugin-app-backend                    0.3.37
  @backstage/plugin-auth-backend                   0.17.0
  @backstage/plugin-auth-node                      0.2.6
  @backstage/plugin-catalog-backend                1.5.0
  @backstage/plugin-catalog-common                 1.0.7
  @backstage/plugin-catalog-graph                  0.2.22
  @backstage/plugin-catalog-import                 0.9.0
  @backstage/plugin-catalog-node                   1.2.0
  @backstage/plugin-catalog-react                  1.2.0
  @backstage/plugin-catalog                        1.6.0
  @backstage/plugin-github-actions                 0.5.10
  @backstage/plugin-org                            0.5.10
  @backstage/plugin-permission-common              0.7.0
  @backstage/plugin-permission-node                0.7.0
  @backstage/plugin-permission-react               0.4.6
  @backstage/plugin-proxy-backend                  0.2.31
  @backstage/plugin-scaffolder-backend             1.7.0
  @backstage/plugin-scaffolder-common              1.2.1
  @backstage/plugin-scaffolder                     1.7.0
  @backstage/plugin-search-backend-module-pg       0.4.1
  @backstage/plugin-search-backend-node            1.0.3
  @backstage/plugin-search-backend                 1.1.0
  @backstage/plugin-search-common                  1.1.0
  @backstage/plugin-search-react                   1.2.0
  @backstage/plugin-search                         1.0.3
  @backstage/plugin-tech-radar                     0.5.17
  @backstage/plugin-techdocs-backend               1.4.0
  @backstage/plugin-techdocs-module-addons-contrib 1.0.5
  @backstage/plugin-techdocs-node                  1.4.1
  @backstage/plugin-techdocs-react                 1.0.5
  @backstage/plugin-techdocs                       1.3.3
  @backstage/plugin-user-settings                  0.5.0
  @backstage/release-manifests                     0.0.6
  @backstage/test-utils                            1.2.1
  @backstage/theme                                 0.2.16
  @backstage/types                                 1.0.0
  @backstage/version-bridge                        1.0.1
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by jhaals** [comment]
Hi @mmacdonald86, sorry to hear that you're having issues! The bug you explain should be fixed in the latest release, I wonder if there might be something that cache the npx command because I was not able to reproduce this locally.
The quick fix if you want to just carry on is to `git init` inside the directory. You can also try to `npx @backstage/create-app@latest` which should hopefully give you the version where this is fixed.

**Comment by freben** [comment]
Closing as fixed - please let us know if this was not correct!

**Comment by steve-r-carter** [comment]
I encountered this today with the latest versions. I performed a git init and that just changed the error to :
[0] fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
[0] Use '--' to separate paths from revisions, like this:
[0] 'git <command> [<revision>...] -- [<file>...]'
[0] fatal: Not a valid object name HEAD
[0] NOTE: Did not compute git version or commit hash, could not execute the git command line utility

my question is what is this part of the startup trying to accomplish? I can access the web UI but I got another error in UI "Failed to load entity kinds" which may or may not be related. I will open a separate ticket for that if I cannot resolve.
